ui/job-view/JobView.jsx

```diff
@@ -264,6 +264,7 @@ class JobView extends React.Component {
                   revision={revision}
                   currentRepo={currentRepo}
                   filterModel={filterModel}
+                  history={this.history}
```

Comment:
It was hard to tell from my quick glance at the history package docs, but does it capture state in the way we're using it at the moment, or just offer a nice API? 

If the latter, do we still need to pass it around as props, or can we just import it at point of use? Or if it does maintain state, and so we need to re-use it - is the new React context API meant for this type of case? (Reading https://reactjs.org/docs/context.html I can't tell for sure)

Reply:
This is a really good question.  I've experimented with that concept of just creating a new ``history`` object in-place.  The listening works fine with separate instances.  I was also using it to manipulate the url, but in more investigation, I don't really need to do that.  I simplified this in ``FilterModel`` so that it just sets ``location.hash`` to the new value.  This works really well.  So then I realized we were just using ``history`` to listen for changes.

But we don't need to now.  Since we're using the hash urls, I can use ``window.addEventListener('hashchange', func, false)``.  And when we move away from using hashes, we can listen for ``popstate`` instead.  So I'm not thinking we need to use the ``history`` package at all right now.  Perhaps if we discover that it gives us some feature we don't easily get by the base packages, then it'll be worth using.

Thanks for making me think this through further.